api/api/src/main/scala/uk/ac/wellcome/platform/api/elasticsearch/ElasticsearchQuery.scala

```diff
@@ -2,35 +2,104 @@ package uk.ac.wellcome.platform.api.elasticsearch
 
 import com.sksamuel.elastic4s.ElasticDsl.{must, should}
 import com.sksamuel.elastic4s.requests.common.Operator
-import com.sksamuel.elastic4s.requests.searches.queries.{ConstantScore, Query}
+import com.sksamuel.elastic4s.requests.searches.queries.Query
 import com.sksamuel.elastic4s.requests.searches.queries.matches.{
   FieldWithOptionalBoost,
   MatchQuery,
   MultiMatchQuery,
   MultiMatchQueryBuilderType
 }
 
+/**
+  * We have `ComboQuery`s and `PartialQuery`s.
+  *
+  * `ComboQuery`s are what we use in the application as they're a
+  * whole set of `PartialQuery`s stuck together, potentially with
+  * different boosting etc.
+  *
+  * These are all quite hard to test _effectively_, as TF/IDF and
+  * all the other amazing Elastic stuff means we land up massaging
+  * the test data to eventually work or leaving the tests out.
+  *
+  * We do have core functionality tests that will run on all available
+  * `ComboQuery` but will use reporting to let us know how they are
+  * working for the public.
+  */
 sealed trait ElasticsearchQuery {
   val q: String
   val elasticQuery: Query
 }
 
-// Queries sent to the application _should_ implement this class
-// as it wraps core functionality that we wouldn't want to break.
-// We haven't enforced this in the type system as we _might_ want
-// to try more extreme tests.
+trait ElasticsearchPartialQuery extends ElasticsearchQuery
+trait ElasticsearchComboQuery extends ElasticsearchQuery
```

Comment:
Are these traits actually used anywhere? It looks like they're just used for tagging different query types, but I can't find anywhere where those tags are used

Reply:
Ah yes - this is exactly what I was trying to enforce via the type system and forgot.
https://github.com/wellcomecollection/catalogue/pull/459/commits/b54074ddc4abe2ddd603723103fb9dde8a4b835e

Tests didn't pick it up as the previous queries had the core query in, and they did the core amount of work, which is great.